crates/blockifier/src/execution/native_syscall_handler.rs

```diff
@@ -56,9 +56,11 @@ impl<'state> StarkNetSyscallHandler for NativeSyscallHandler<'state> {
         _remaining_gas: &mut u128,
     ) -> SyscallResult<Felt> {
         if self.execution_context.execution_mode == ExecutionMode::Validate {
-            let execution_mode_err = Felt::from_hex(INVALID_EXECUTION_MODE_ERROR).unwrap();
+            let execution_mode_err = encode_str_as_felts(
+                "Unauthorized syscall get_block_hash in execution mode Validate",
```

Comment:
It can cause problems in the future to use directly the string. Let's use the same error type the Blockifier uses, and convert it to string:

```rs
let err = SyscallExecutionError::InvalidSyscallInExecutionMode {
      syscall_name: "get_block_hash".to_string(),
      execution_mode: syscall_handler.execution_mode(),
 });
 encode_str_as_felts(&err.to_string())
 ```
 
 This way if the error changes, this one does as well